deploy/terraform/modules/anydb_node/variables_local.tf

```
@@ -0,0 +1,160 @@
+
+  # Imports database sizing information
+  sizes = jsondecode(file("${path.root}/../anydb_sizes.json"))
+
+  # PPG Information
+  ppgId = lookup(var.infrastructure, "ppg", false) != false ? (var.ppg[0].id) : null
+
+  # Filter the list of databases to only AnyDB platform entries
+  # Supported databases: Oracle, DB2, SQLServer, ASE 
+  anydb-databases = [
+    for database in var.databases : database
+    if contains(["ORACLE", "DB2", "SQLSERVER", "ASE"], upper(try(database.platform, "NONE")))
+  ]
+
+  # if(upper(database.platform) in )
+  # Enable deployment based on length of local.anydb-databases
+  enable_deployment = (length(local.anydb-databases) > 0) ? true : false
+
+  anydb          = try(local.anydb-databases[0], {})
+  anydb_platform = try(local.anydb.platform, "NONE")
+  anydb_version  = try(local.anydb.db_version, "")
+
+  # OS image for all Application Tier VMs
+  # If custom image is used, we do not overwrite os reference with default value
+  anydb_custom_image = try(local.anydb.os.source_image_id, "") != "" ? true : false
+
+  anydb_os = {
+    "source_image_id" = local.anydb_custom_image ? local.anydb.os.source_image_id : ""
+    "publisher"       = try(local.anydb.os.publisher, local.anydb_custom_image ? "" : "")
+    "offer"           = try(local.anydb.os.offer, local.anydb_custom_image ? "" : "")
+    "sku"             = try(local.anydb.os.sku, local.anydb_custom_image ? "" : "")
+    "version"         = try(local.anydb.os.version, local.anydb_custom_image ? "" : "latest")
+  }
+
+  anydb_ostype = try(local.anydb.os.os_type, "Linux")
+  anydb_size   = try(local.anydb.size, "500")
+  anydb_sku    = try(lookup(local.sizes, local.anydb_size).compute.vmsize,"Standard_E4s_v3")
+  anydb_fs     = try(local.anydb.filesystem, "xfs")
+  anydb_ha     = try(local.anydb.high_availability, "false")
+  anydb_sid    = (length(local.anydb-databases) > 0) ? try(local.anydb.instance.sid, "OR1") : "OR1"
+  loadbalancer = try(local.anydb.loadbalancer, {})
+
+  authentication = try(var.application.authentication,
+    {
+      "type"     = upper(local.anydb_ostype) == "LINUX" ? "key" : "password"
+      "username" = "azureadm"
+      "password" = "Sap@hana2019!"
+  })
+
+  # Update database information with defaults
+  anydb_database = merge(local.anydb,
+    { platform = local.anydb_platform },
+    { db_version = local.anydb_version },
+    { size = local.anydb_size },
+    { os = local.anydb_ostype },
+    { filesystem = local.anydb_fs },
+    { high_availability = local.anydb_ha },
+    { authentication = local.authentication }
+  )
+
+  dbnodes = flatten([
+    [
+      for database in local.anydb-databases : [
+        for idx, dbnode in database.dbnodes : {
+          platform       = local.anydb_platform,
+          name           = upper(local.anydb_ostype) == "WINDOWS" ? format("%s_xdbw%02d-0", upper(local.anydb_sid), idx) : format("%s_xdbl%02d-0", upper(local.anydb_sid), idx),
```

Sorry overlooked this earlier, the logic would be if name not provided, use naming convention:
```suggestion
          name           = try(dbnode.name, upper(local.anydb_ostype) == "WINDOWS" ? format("%s_xdbw%02d-0", upper(local.anydb_sid), idx) : format("%s_xdbl%02d-0", upper(local.anydb_sid), idx)),
```

---

fixed